docs/KMP searching pattern/readme.md

```diff
@@ -0,0 +1,156 @@
+---
+id: kmp-searching-pattern
+title: KMP searching pattern
+sidebar_label: KMP Searching
+description: This algorithm is an efficient string matching that finds the different occurence of a word.
+tags: [Dynamic Programming,Algorithm, DSA]
+---
+
+
+# 🔍 KMP (Knuth-Morris-Pratt) Pattern Searching Algorithm
+
+## 📚 Overview
+The KMP algorithm is an efficient string matching algorithm that finds occurrences of a "word" W within a main "text string" S by employing the observation that when a mismatch occurs, the word itself embodies sufficient information to determine where the next match could begin, thus bypassing re-examination of previously matched characters.
+
+<div align="center">
+    <img src="https://media.geeksforgeeks.org/wp-content/uploads/20221125004358/image-660x398.png" alt="KMP Algorithm" width="500">
```

Comment:
<img> is a non-void element. It requires a closing tag. Replace `<img>` with `<img />` to make it self-closing.